### PR TITLE
standardize target platform override behavior in tool and update gallery to support it

### LINKED ISSUE #23902: Make debugTogglePlatform toggle to fuchsia too
This might only need to take effect when connected to a fuchsia device

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Standardize the behavior of the default target platform override to move from android, iOS, fuchsia, (macOS if enabled). Fixes the gallery so that updates to this setting from the command line show in the application. 

Fixes https://github.com/flutter/flutter/issues/23902

### CODE REVIEW

File: examples/flutter_gallery/lib/gallery/app.dart [hunk 1]
```diff
@@ -70,6 +70,12 @@ class _GalleryAppState extends State<GalleryApp> {
     model = AppStateModel()..loadProducts();
   }
 
+  @override
+  void reassemble() {
+    _options = _options.copyWith(platform: defaultTargetPlatform);
```
Comment (jonahwilliams) [inline examples/flutter_gallery/lib/gallery/app.dart]: Without this, the options does not get updated for new platforms on a reassemble

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 2]
```diff
@@ -1299,3 +1290,25 @@ class DebugConnectionInfo {
...
+    case 'iOS':
+      return 'fuchsia';
+    case 'fuchsia':
+      if (featureFlags.isMacOSEnabled) {
+        return 'macOS';
+      }
+      return 'android';
+    case 'macOS':
+      return 'android';
+    default:
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/resident_runner.dart]: What does it mean `currentPlatform` was something else? Should we emit a warning here?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_runner.dart]: Perhaps there was a bug in the serialization code on the framework side, or a bug else where. Maybe we could assert here?

**Review by christopherfujino — APPROVED** [review]
LGTM, with a nit question.

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_runner.dart]: Updated to use an assert

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/resident_runner.dart]: LGTM

**Pull request merged**